### ISSUE TITLE
Incompatibility with Sodium 0.6.1

### ISSUE THREAD

**Issue by jurohn** [Wurst-Imperium/Wurst7#1088]
### Pre-reporting checklist

- [X] I have searched existing issues and didn't find any previous reports of this bug.
- [X] This bug still happens in the latest version of Wurst. I've made sure my installation is up to date.
- [X] I have tried removing all other mods and the bug still happens.

### Description

Open a random world (smp) with this mods and it crashes.

### Repro steps

Open a random world (smp) with this mods and it crashes. Didnt crash with sodium 0.6 but they say they dont support Wurst.

![391951883-62a7cb90-85b7-4c25-ae1d-e124d861d7af](https://github.com/user-attachments/assets/faf6bb9b-d264-4a5a-8595-2e99ad28b60f)

### Wurst Client version

7.46.3

### Minecraft version

1.21.3

### Crash report (if applicable)

```plaintext
---- Minecraft Crash Report ----
// Hi. I'm Minecraft, and I'm a crashaholic.

Time: 2024-12-03 12:47:03
Description: Unexpected error

java.lang.NullPointerException: Cannot invoke "net.minecraft.class_1297.method_5765()" because "$$1" is null
	at knot//net.minecraft.class_4184.method_19321(class_4184.java:57)
	at knot//net.minecraft.class_757.method_3188(class_757.java:690)
	at knot//net.minecraft.class_757.method_3192(class_757.java:491)
	at knot//net.minecraft.class_310.method_1523(class_310.java:1352)
	at knot//net.minecraft.class_310.method_1514(class_310.java:933)
	at knot//net.minecraft.client.main.Main.main(Main.java:265)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:480)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Render thread
Stacktrace:
	at knot//net.minecraft.class_4184.method_19321(class_4184.java:57)
	at knot//net.minecraft.class_757.method_3188(class_757.java:690)

-- Uptime --
Details:
	JVM uptime: 18.790s
	Wall uptime: 10.970s
	High-res time: 8.754s
	Client ticks: 138 ticks / 6.900s
Stacktrace:
	at knot//net.minecraft.class_310.method_1587(class_310.java:2443)
	at knot//net.minecraft.class_310.method_54580(class_310.java:1002)
	at knot//net.minecraft.class_310.method_1514(class_310.java:957)
	at knot//net.minecraft.client.main.Main.main(Main.java:265)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:480)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)

-- Affected level --
Details:
	All players: 0 total; []
	Chunk stats: 361, 0
	Level dimension: minecraft:worldhaha
	Level spawn location: World: (8,64,8), Section: (at 8,0,8 in 0,4,0; chunk contains blocks 0,-64,0 to 15,319,15), Region: (0,0; contains chunks 0,0 to 31,31, blocks 0,-64,0 to 511,319,511)
	Level time: 0 game time, 0 day time
	Server brand: ~~ERROR~~ NullPointerException: Cannot read field "field_3944" because "this.field_3729.field_1724" is null
	Server type: Non-integrated multiplayer server
	Tracked entity count: 0

-- Last reload --
Details:
	Reload number: 1
	Reload reason: initial
	Finished: Yes
	Packs: vanilla, fabric, appleskin, betteradvancements, c2me, capes, chat_heads, chatanimation, chatnotify, cicada, clean-debug, cloth-config, compact-chat, concentration, detailab, effecttimerplus, entity_model_features, entity_texture_features, fabric-api, fabric-api-base, fabric-api-lookup-api-v1, fabric-biome-api-v1, fabric-block-api-v1, fabric-block-view-api-v2, fabric-blockrenderlayer-v1, fabric-client-tags-api-v1, fabric-command-api-v1, fabric-command-api-v2, fabric-commands-v0, fabric-content-registries-v0, fabric-convention-tags-v1, fabric-convention-tags-v2, fabric-crash-report-info-v1, fabric-data-attachment-api-v1, fabric-data-generation-api-v1, fabric-dimensions-v1, fabric-entity-events-v1, fabric-events-interaction-v0, fabric-game-rule-api-v1, fabric-item-api-v1, fabric-item-group-api-v1, fabric-key-binding-api-v1, fabric-keybindings-v0, fabric-language-kotlin, fabric-lifecycle-events-v1, fabric-loot-api-v2, fabric-loot-api-v3, fabric-message-api-v1, fabric-model-loading-api-v1, fabric-networking-api-v1, fabric-object-builder-api-v1, fabric-particles-v1, fabric-recipe-api-v1, fabric-registry-sync-v0, fabric-renderer-api-v1, fabric-renderer-indigo, fabric-rendering-data-attachment-v1, fabric-rendering-fluids-v1, fabric-rendering-v0, fabric-rendering-v1, fabric-resource-conditions-api-v1, fabric-resource-loader-v0, fabric-screen-api-v1, fabric-screen-handler-api-v1, fabric-sound-api-v1, fabric-transfer-api-v1, fabric-transitive-access-wideners-v1, fabricloader, fadeless, fallingleaves, forcecloseworldloadingscreen, hswf3, immediatelyfast, krypton, lithium, midnightlib, modmenu, morechathistory, reeses-sodium-options, rrls, scalablelux, shulkerboxtooltip, skinlayers3d, sodium, sodium-extra, tier-tagger, timestamp_chat, ukulib, ukus-armor-hud, voicechat, wurst, yet_another_config_lib_v3, file/Blue Ancient Debris 1.20.zip

-- System Details --
Details:
	Minecraft Version: 1.21.3
	Minecraft Version ID: 1.21.3
	Operating System: Windows 11 (amd64) version 10.0
	Java Version: 21.0.3, Microsoft
	Java VM Version: OpenJDK 64-Bit Server VM (mixed mode), Microsoft
	Memory: 419478200 bytes (400 MiB) / 805306368 bytes (768 MiB) up to 2147483648 bytes (2048 MiB)
	CPUs: 16
	Processor Vendor: AuthenticAMD
	Processor Name: AMD Ryzen 7 7700X 8-Core Processor             
	Identifier: AuthenticAMD Family 25 Model 97 Stepping 2
	Microarchitecture: Zen 3
	Frequency (GHz): 4.49
	Number of physical packages: 1
	Number of physical CPUs: 8
	Number of logical CPUs: 16
	Graphics card #0 name: AMD Radeon(TM) Graphics
	Graphics card #0 vendor: Advanced Micro Devices, Inc.
	Graphics card #0 VRAM (MiB): 512.00
	Graphics card #0 deviceId: VideoController1
	Graphics card #0 versionInfo: 32.0.11024.2
	Graphics card #1 name: NVIDIA GeForce RTX 3060 Ti
	Graphics card #1 vendor: NVIDIA
	Graphics card #1 VRAM (MiB): 8192.00
	Graphics card #1 deviceId: VideoController2
	Graphics card #1 versionInfo: 32.0.15.6614
	Memory slot #0 capacity (MiB): 16384.00
	Memory slot #0 clockSpeed (GHz): 4.80
	Memory slot #0 type: Unknown
	Memory slot #1 capacity (MiB): 16384.00
	Memory slot #1 clockSpeed (GHz): 4.80
	Memory slot #1 type: Unknown
	Virtual memory max (MiB): 40597.48
	Virtual memory used (MiB): 19415.37
	Swap memory total (MiB): 8704.00
	Swap memory used (MiB): 16.81
	Space in storage for jna.tmpdir (MiB): available: 864511.31, total: 1906800.00
	Space in storage for org.lwjgl.system.SharedLibraryExtractPath (MiB): available: 864511.31, total: 1906800.00
	Space in storage for io.netty.native.workdir (MiB): available: 864511.31, total: 1906800.00
	Space in storage for java.io.tmpdir (MiB): available: 864511.31, total: 1906800.00
	Space in storage for workdir (MiB): available: 864511.31, total: 1906800.00
	JVM Flags: 9 total; -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump -Xss1M -Xmx2G -XX:+UnlockExperimentalVMOptions -XX:+UseG1GC -XX:G1NewSizePercent=20 -XX:G1ReservePercent=20 -XX:MaxGCPauseMillis=50 -XX:G1HeapRegionSize=32M
	Fabric Mods: 
		appleskin: AppleSkin 3.0.6+mc1.21.3
		architectury: Architectury 14.0.3
		betteradvancements: Better Advancements 0.4.4.22
		c2me: Concurrent Chunk Management Engine 0.3.0.0.0+1.21.3
			c2me-base: Concurrent Chunk Management Engine (Base) 0.3.0.0.0+1.21.3
			c2me-client-uncapvd: Concurrent Chunk Management Engine (Client/Uncap View Distance) 0.3.0.0.0+1.21.3
			c2me-fixes-chunkio-threading-issues: Concurrent Chunk Management Engine (Fixes/Chunk IO/Threading Issues) 0.3.0.0.0+1.21.3
			c2me-fixes-general-threading-issues: Concurrent Chunk Management Engine (Fixes/General/Threading Issues) 0.3.0.0.0+1.21.3
			c2me-fixes-worldgen-threading-issues: Concurrent Chunk Management Engine (Fixes/WorldGen/Threading Issues) 0.3.0.0.0+1.21.3
			c2me-fixes-worldgen-vanilla-bugs: Concurrent Chunk Management Engine (Fixes/WorldGen/Vanilla Bugs) 0.3.0.0.0+1.21.3
			c2me-notickvd: Concurrent Chunk Management Engine (No Tick View Distance) 0.3.0.0.0+1.21.3
			c2me-opts-allocs: Concurrent Chunk Management Engine (Optimizations/Memory Allocations) 0.3.0.0.0+1.21.3
			c2me-opts-chunk-access: Concurrent Chunk Management Engine (Optimizations/Chunk Access) 0.3.0.0.0+1.21.3
			c2me-opts-chunkio: Concurrent Chunk Management Engine (Optimizations/Chunk IO) 0.3.0.0.0+1.21.3
			c2me-opts-dfc: Concurrent Chunk Management Engine (Optimizations/Density Function Compiler) 0.3.0.0.0+1.21.3
			c2me-opts-math: Concurrent Chunk Management Engine (Optimizations/Math) 0.3.0.0.0+1.21.3
			c2me-opts-scheduling: Concurrent Chunk Management Engine (Optimizations/Scheduling) 0.3.0.0.0+1.21.3
			c2me-opts-worldgen-general: Concurrent Chunk Management Engine (Optimizations/General WorldGen) 0.3.0.0.0+1.21.3
			c2me-opts-worldgen-vanilla: Concurrent Chunk Management Engine (Optimizations/Vanilla WorldGen) 0.3.0.0.0+1.21.3
			c2me-rewrites-chunk-serializer: Concurrent Chunk Management Engine (Rewrites/Chunk Serializer) 0.3.0.0.0+1.21.3
			c2me-rewrites-chunk-system: Concurrent Chunk Management Engine (Rewrites/Chunk System) 0.3.0.0.0+1.21.3
			c2me-rewrites-chunkio: Concurrent Chunk Management Engine (Rewrites/Chunk IO) 0.3.0.0.0+1.21.3
			c2me-server-utils: Concurrent Chunk Management Engine (Server Utils) 0.3.0.0.0+1.21.3
			c2me-threading-lighting: Concurrent Chunk Management Engine (Threading/Lighting) 0.3.0.0.0+1.21.3
			com_electronwill_night-config_core: core 3.6.5
			com_electronwill_night-config_toml: toml 3.6.5
			com_ibm_async_asyncutil: asyncutil 0.1.0
			io_reactivex_rxjava3_rxjava: rxjava 3.1.8
			mixinsquared: MixinSquared 0.2.0-beta.6
			net_objecthunter_exp4j: exp4j 0.4.8
			org_jctools_jctools-core: jctools-core 4.0.5
			org_reactivestreams_reactive-streams: reactive-streams 1.0.4
		capes: Capes 1.5.4+1.21.2
		chat_heads: Chat Heads 0.13.6
		chatanimation: ChatAnimation 1.0.6
		chatnotify: ChatNotify 2.2.0+1.21.3
		cicada: CICADA 0.8.4-beta.1+1.21.2-and-above
		clean-debug: Clean F3 0.4.8
		cloth-config: Cloth Config v16 16.0.141
			cloth-basic-math: cloth-basic-math 0.6.1
		compact-chat: Compact Chat 2.1.1
		concentration: Concentration 2.3.0
		dcch: DCCH 1.2
		detailab: Detail Armor Bar 2.6.3+1.21.3-fabric
		effecttimerplus: EffectTimerPlus 2.2.5+1.21.3
		entity_model_features: Entity Model Features 2.3.0
		entity_texture_features: Entity Texture Features 6.2.8
			org_apache_httpcomponents_httpmime: httpmime 4.5.10
		fabric-api: Fabric API 0.110.0+1.21.3
			fabric-api-base: Fabric API Base 0.4.50+119c825640
			fabric-api-lookup-api-v1: Fabric API Lookup API (v1) 1.6.81+fd37071f40
			fabric-biome-api-v1: Fabric Biome API (v1) 14.0.6+fd37071f40
			fabric-block-api-v1: Fabric Block API (v1) 1.0.28+fd37071f40
			fabric-block-view-api-v2: Fabric BlockView API (v2) 1.0.16+fd37071f40
			fabric-blockrenderlayer-v1: Fabric BlockRenderLayer Registration (v1) 2.0.4+fd37071f40
			fabric-client-tags-api-v1: Fabric Client Tags 1.1.24+fd37071f40
			fabric-command-api-v1: Fabric Command API (v1) 1.2.58+f71b366f40
			fabric-command-api-v2: Fabric Command API (v2) 2.2.37+c9d82ab240
			fabric-commands-v0: Fabric Commands (v0) 0.2.75+df3654b340
			fabric-content-registries-v0: Fabric Content Registries (v0) 9.1.8+fd37071f40
			fabric-convention-tags-v1: Fabric Convention Tags 2.1.3+7f945d5b40
			fabric-convention-tags-v2: Fabric Convention Tags (v2) 2.9.3+84ebbb0e40
			fabric-crash-report-info-v1: Fabric Crash Report Info (v1) 0.3.3+fd37071f40
			fabric-data-attachment-api-v1: Fabric Data Attachment API (v1) 1.4.0+dbad41f140
			fabric-data-generation-api-v1: Fabric Data Generation API (v1) 21.0.18+fd37071f40
			fabric-dimensions-v1: Fabric Dimensions API (v1) 4.0.7+fd37071f40
			fabric-entity-events-v1: Fabric Entity Events (v1) 2.0.8+825023f840
			fabric-events-interaction-v0: Fabric Events Interaction (v0) 2.0.12+c2c3ac4740
			fabric-game-rule-api-v1: Fabric Game Rule API (v1) 1.0.59+fd37071f40
			fabric-item-api-v1: Fabric Item API (v1) 11.1.7+fd37071f40
			fabric-item-group-api-v1: Fabric Item Group API (v1) 4.1.17+fd37071f40
			fabric-key-binding-api-v1: Fabric Key Binding API (v1) 1.0.53+fd37071f40
			fabric-keybindings-v0: Fabric Key Bindings (v0) 0.2.51+df3654b340
			fabric-lifecycle-events-v1: Fabric Lifecycle Events (v1) 2.5.1+6da5ef6940
			fabric-loot-api-v2: Fabric Loot API (v2) 3.0.26+3f89f5a540
			fabric-loot-api-v3: Fabric Loot API (v3) 1.0.14+fd37071f40
			fabric-message-api-v1: Fabric Message API (v1) 6.0.21+fd37071f40
			fabric-model-loading-api-v1: Fabric Model Loading API (v1) 3.0.9+fd37071f40
			fabric-networking-api-v1: Fabric Networking API (v1) 4.3.5+fd37071f40
			fabric-object-builder-api-v1: Fabric Object Builder API (v1) 17.0.9+fd37071f40
			fabric-particles-v1: Fabric Particles (v1) 4.0.10+fd37071f40
			fabric-recipe-api-v1: Fabric Recipe API (v1) 7.0.4+fd37071f40
			fabric-registry-sync-v0: Fabric Registry Sync (v0) 5.3.0+2758bfbf40
			fabric-renderer-api-v1: Fabric Renderer API (v1) 4.0.5+fd37071f40
			fabric-renderer-indigo: Fabric Renderer - Indigo 1.9.5+fd37071f40
			fabric-rendering-data-attachment-v1: Fabric Rendering Data Attachment (v1) 0.3.54+73761d2e40
			fabric-rendering-fluids-v1: Fabric Rendering Fluids (v1) 3.1.15+fd37071f40
			fabric-rendering-v0: Fabric Rendering (v0) 1.1.84+df3654b340
			fabric-rendering-v1: Fabric Rendering (v1) 8.0.8+6922831640
			fabric-resource-conditions-api-v1: Fabric Resource Conditions API (v1) 5.0.8+fd37071f40
			fabric-resource-loader-v0: Fabric Resource Loader (v0) 3.0.6+fd37071f40
			fabric-screen-api-v1: Fabric Screen API (v1) 2.0.34+fd37071f40
			fabric-screen-handler-api-v1: Fabric Screen Handler API (v1) 1.3.101+fd37071f40
			fabric-sound-api-v1: Fabric Sound API (v1) 1.0.29+fd37071f40
			fabric-transfer-api-v1: Fabric Transfer API (v1) 5.4.3+fd37071f40
			fabric-transitive-access-wideners-v1: Fabric Transitive Access Wideners (v1) 6.2.0+54a41b1c40
		fabric-language-kotlin: Fabric Language Kotlin 1.13.0+kotlin.2.1.0
			org_jetbrains_kotlin_kotlin-reflect: kotlin-reflect 2.1.0
			org_jetbrains_kotlin_kotlin-stdlib: kotlin-stdlib 2.1.0
			org_jetbrains_kotlin_kotlin-stdlib-jdk7: kotlin-stdlib-jdk7 2.1.0
			org_jetbrains_kotlin_kotlin-stdlib-jdk8: kotlin-stdlib-jdk8 2.1.0
			org_jetbrains_kotlinx_atomicfu-jvm: atomicfu-jvm 0.26.1
			org_jetbrains_kotlinx_kotlinx-coroutines-core-jvm: kotlinx-coroutines-core-jvm 1.9.0
			org_jetbrains_kotlinx_kotlinx-coroutines-jdk8: kotlinx-coroutines-jdk8 1.9.0
			org_jetbrains_kotlinx_kotlinx-datetime-jvm: kotlinx-datetime-jvm 0.6.1
			org_jetbrains_kotlinx_kotlinx-io-bytestring-jvm: kotlinx-io-bytestring-jvm 0.6.0
			org_jetbrains_kotlinx_kotlinx-io-core-jvm: kotlinx-io-core-jvm 0.6.0
			org_jetbrains_kotlinx_kotlinx-serialization-cbor-jvm: kotlinx-serialization-cbor-jvm 1.7.3
			org_jetbrains_kotlinx_kotlinx-serialization-core-jvm: kotlinx-serialization-core-jvm 1.7.3
			org_jetbrains_kotlinx_kotlinx-serialization-json-jvm: kotlinx-serialization-json-jvm 1.7.3
		fabricloader: Fabric Loader 0.16.9
			mixinextras: MixinExtras 0.4.1
		fadeless: Fadeless 2.0.1
			midnightlib: MidnightLib 1.6.3
		fallingleaves: Falling Leaves 1.16.4
		ferritecore: FerriteCore 7.1.0-hotfix
		forcecloseworldloadingscreen: kennytv's epic force close loading screen mod for Fabric 2.2.2
		hswf3: Hide Scoreboard When F3 1.0.1
		immediatelyfast: ImmediatelyFast 1.3.2+1.21.3
			net_lenni0451_reflect: Reflect 1.3.4
		java: OpenJDK 64-Bit Server VM 21
		krypton: Krypton 0.2.8
			com_velocitypowered_velocity-native: velocity-native 3.3.0-SNAPSHOT
		lithium: Lithium 0.14.2-snapshot+mc1.21.3-build.91
		minecraft: Minecraft 1.21.3
		modmenu: Mod Menu 12.0.0-beta.1
			placeholder-api: Placeholder API 2.4.0-pre.2+1.21
		morechathistory: More Chat History 1.3.1
		reeses-sodium-options: Reese's Sodium Options 1.8.0+mc1.21.3
		rrls: Remove Reloading Screen 5.0.10+mc1.21.3-fabric
		scalablelux: ScalableLux 0.1.1+fabric.452731d
		shulkerboxtooltip: Shulker Box Tooltip 5.2.1+1.21.3
		skinlayers3d: 3d-Skin-Layers 1.7.2
		sodium: Sodium 0.6.1+mc1.21.3
		sodium-extra: Sodium Extra 0.6.0+mc1.21.3
		tier-tagger: TierTagger 1.12.0+mc1.21.2
		timestamp_chat: Timestamp Chat 2.4.0+1.19.1
		ukulib: ukulib 1.5.0+1.21.2
			com_moandjiezana_toml_toml4j: toml4j 0.7.2
		ukus-armor-hud: uku's Armor HUD 0.6.0+mc1.21.2
		voicechat: Simple Voice Chat 1.21.3-2.5.26
		wurst: Wurst Client v7.46.3-MC1.21.3
		yet_another_config_lib_v3: YetAnotherConfigLib 3.6.1+1.21.2-fabric
			com_twelvemonkeys_common_common-image: common-image 3.10.0
			com_twelvemonkeys_common_common-io: common-io 3.10.0
			com_twelvemonkeys_common_common-lang: common-lang 3.10.0
			com_twelvemonkeys_imageio_imageio-core: imageio-core 3.10.0
			com_twelvemonkeys_imageio_imageio-metadata: imageio-metadata 3.10.0
			com_twelvemonkeys_imageio_imageio-webp: imageio-webp 3.10.0
			org_quiltmc_parsers_gson: gson 0.2.1
			org_quiltmc_parsers_json: json 0.2.1
	Launched Version: fabric-loader-0.16.9-1.21.3
	Launcher name: minecraft-launcher
	Backend library: LWJGL version 3.3.3-snapshot
	Backend API: NVIDIA GeForce RTX 3060 Ti/PCIe/SSE2 GL version 3.2.0 NVIDIA 566.14, NVIDIA Corporation
	Window size: 1278x741
	GFLW Platform: win32
	GL Caps: Using framebuffer using OpenGL 3.2
	GL debug messages: 
	Is Modded: Definitely; Client brand changed to 'fabric'
	Universe: 400921fb54442d18
	Type: Client (map_client.txt)
	Graphics mode: fancy
	Render Distance: 32/32 chunks
	Resource Packs: vanilla, fabric, appleskin (incompatible), betteradvancements (incompatible), c2me, capes, chat_heads, chatanimation, chatnotify (incompatible), cicada, clean-debug, cloth-config, compact-chat, concentration (incompatible), detailab, effecttimerplus (incompatible), entity_model_features, entity_texture_features, fabric-api, fabric-api-base, fabric-api-lookup-api-v1, fabric-biome-api-v1, fabric-block-api-v1, fabric-block-view-api-v2, fabric-blockrenderlayer-v1, fabric-client-tags-api-v1, fabric-command-api-v1, fabric-command-api-v2, fabric-commands-v0, fabric-content-registries-v0, fabric-convention-tags-v1, fabric-convention-tags-v2, fabric-crash-report-info-v1, fabric-data-attachment-api-v1, fabric-data-generation-api-v1, fabric-dimensions-v1, fabric-entity-events-v1, fabric-events-interaction-v0, fabric-game-rule-api-v1, fabric-item-api-v1, fabric-item-group-api-v1, fabric-key-binding-api-v1, fabric-keybindings-v0, fabric-language-kotlin, fabric-lifecycle-events-v1, fabric-loot-api-v2, fabric-loot-api-v3, fabric-message-api-v1, fabric-model-loading-api-v1, fabric-networking-api-v1, fabric-object-builder-api-v1, fabric-particles-v1, fabric-recipe-api-v1, fabric-registry-sync-v0, fabric-renderer-api-v1, fabric-renderer-indigo, fabric-rendering-data-attachment-v1, fabric-rendering-fluids-v1, fabric-rendering-v0, fabric-rendering-v1, fabric-resource-conditions-api-v1, fabric-resource-loader-v0, fabric-screen-api-v1, fabric-screen-handler-api-v1, fabric-sound-api-v1, fabric-transfer-api-v1, fabric-transitive-access-wideners-v1, fabricloader, fadeless, fallingleaves, forcecloseworldloadingscreen, hswf3, immediatelyfast, krypton, lithium, midnightlib, modmenu, morechathistory, reeses-sodium-options, rrls, scalablelux, shulkerboxtooltip, skinlayers3d, sodium, sodium-extra, tier-tagger, timestamp_chat (incompatible), ukulib, ukus-armor-hud, voicechat, wurst, yet_another_config_lib_v3 (incompatible), file/Blue Ancient Debris 1.20.zip (incompatible)
	Current Language: en_us
	Locale: en_US
	System encoding: Cp1252
	File encoding: UTF-8
	CPU: 16x AMD Ryzen 7 7700X 8-Core Processor
```

**Comment by Alexander01998** [comment]
Deobfuscated stacktrace:

```
java.lang.NullPointerException: Cannot invoke "net.minecraft.entity.Entity.hasVehicle" because "$$1" is null
	at knot//net.minecraft.client.render.Camera.update(Camera.java:57)
	at knot//net.minecraft.client.render.GameRenderer.renderWorld(GameRenderer.java:690)
	at knot//net.minecraft.client.render.GameRenderer.render(GameRenderer.java:491)
	at knot//net.minecraft.client.MinecraftClient.render(MinecraftClient.java:1352)
	at knot//net.minecraft.client.MinecraftClient.run(MinecraftClient.java:933)
	at knot//net.minecraft.client.main.Main.main(Main.java:265)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:480)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)
```

This is a rather cryptic error message. Wurst's mixins do not target `Entity.hasVehicle()` or `Camera.update()` directly. I'm not sure what `$$1` points to.

@jurohn Just to confirm, you did also try it without that huge list of mods and still got the same error with just Wurst, Fabric API, and Sodium?

**Comment by jurohn** [comment]
> Deobfuscated stacktrace:
> 
> ```
> java.lang.NullPointerException: Cannot invoke "net.minecraft.entity.Entity.hasVehicle" because "$$1" is null
> 	at knot//net.minecraft.client.render.Camera.update(Camera.java:57)
> 	at knot//net.minecraft.client.render.GameRenderer.renderWorld(GameRenderer.java:690)
> 	at knot//net.minecraft.client.render.GameRenderer.render(GameRenderer.java:491)
> 	at knot//net.minecraft.client.MinecraftClient.render(MinecraftClient.java:1352)
> 	at knot//net.minecraft.client.MinecraftClient.run(MinecraftClient.java:933)
> 	at knot//net.minecraft.client.main.Main.main(Main.java:265)
> 	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:480)
> 	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
> 	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)
> ```
> 
> This is a rather cryptic error message. Wurst's mixins do not target `Entity.hasVehicle()` or `Camera.update()` directly. I'm not sure what `$$1` points to.
> 
> @jurohn Just to confirm, you did also try it without that huge list of mods and still got the same error with just Wurst, Fabric API, and Sodium?

![image](https://github.com/user-attachments/assets/92f3323c-7c2d-405c-8e08-b65a334cb64d)

Tested with just this and the problem is still there.
Problem was not there with the previous Sodium version, so they must have changed something that messed up Wurst compatibility.

**Comment by MCRcortex** [comment]
the issue is your SodiumFluidRendererMixin

**Comment by Alexander01998** [comment]
Hmm.

`Entity.hasVehicle()` is normally just `return this.getVehicle() != null;`, which should not be able to throw a `NullPointerException` unless it's been modified by a mixin.

Sodium 0.6.1 does not seem to have any mixins targeting `Entity.hasVehicle()`.

Reese's Sodium Options does not seem to have any mixins targeting `Entity` at all.

Sodium Extra does not seem to have any mixins targeting `Entity.hasVehicle()`, but does have a mixin for `Camera.tick()`.

`Camera.tick()` is mojmap for `Camera.updateEyeHeight()`, not `Camera.update()`.

**Comment by Alexander01998** [comment]
> the issue is your SodiumFluidRendererMixin

Please elaborate.

**Comment by MCRcortex** [comment]
The issue is a mixin issue, the SodiumFluidRenderer class gets loaded in on a packet join, due to initialization of the renderer, the exception is caught as all packet errors are and ignored, however this is completely incorrect and results in the incorrect crash file being generated, if you look at the log, its the first error that happened, `Invalid descriptor on wurst.mixins.json:SodiumFluidRendererMixin from mod wurst->@Inject::onIsFluidOccluded`

**Comment by Alexander01998** [comment]
Thanks! I see the issue. The parameters of Sodium's `DefaultFluidRenderer.isFluidOccluded()` method changed in [this commit](https://github.com/CaffeineMC/sodium/commit/87c1bd5aa204ea0a9baef4a11fbd6fd09973a3d8#diff-8ec8b1bb76169fab8cc17c693137e5f013a7e53ae941fedff1a22787dbbb44e9L68-R68) from last week. The method now takes a `FluidState` instead of a `Fluid`, but Wurst is still looking for the old method signature.

**Comment by Alexander01998** [comment]
Will be fixed in Wurst 7.46.4. Let's see if I can get that released before Minecraft 1.21.4 is out.

**Comment by jurohn** [comment]
Thank you @Alexander01998 for picking this up insanely fast. Great!
